packages/read-model/src/Product/Listed/ListedProductViewFacade.php

```diff
@@ -223,6 +223,42 @@ protected function createPaginationResultWithData(PaginationResult $paginationRe
         );
     }
 
+    /**
+     * @param \Shopsys\FrameworkBundle\Component\Paginator\PaginationResult $paginationResult
+     * @return \Shopsys\FrameworkBundle\Component\Paginator\PaginationResult
+     */
+    protected function createPaginationResultWithHits(PaginationResult $paginationResult): PaginationResult
+    {
+        return new PaginationResult(
+            $paginationResult->getPage(),
+            $paginationResult->getPageSize(),
+            $paginationResult->getTotalCount(),
+            $this->createFromHits($paginationResult->getResults())
+        );
+    }
```

Comment:
I understand why and it's absolutely same usage as we introduce and with second usage it feels just wrong :D it's nothing you need to change, just need to mention thoughts after some time :)